lib/index.js

```diff
@@ -0,0 +1,93 @@
+var Promise = require('bluebird');
+var Redis = require('redis-fast-driver');
+var transformations = require('./redis-result-transformation-functions');
+
+var juttle_utils = require('juttle/lib/runtime').utils;
+
+function RedisBackend(config, JuttleRuntime) {
+    var redis = Promise.promisifyAll(new Redis(config));
+    var connect_promise = new Promise(function(resolve, reject) {
+        redis.on('ready', resolve);
+        redis.on('error', reject);
+    });
+
+    var Read = JuttleRuntime.proc.base.extend({
+        procName: 'redis_read',
+        sourceType: 'batch',
+
+        initialize: function(options, params) {
+            if (!options.hasOwnProperty('raw')) {
+                throw new Error('must specify a redis query with -raw');
+            }
+
+            this.command = options.raw.split(' ');
+        },
+
+        start: function() {
+            var self = this;
+            var keyword = this.command[0].toLowerCase();
+
+            if (!transformations.hasOwnProperty(keyword)) {
+                return self.trigger('error', new Error('unsupported/invalid Redis command: ', keyword));
+            }
+            return connect_promise.then(function() {
+                return redis.rawCallAsync(self.command);
+            })
+            .then(function(result) {
+                var points = transformations[keyword](self.command, result);
+
+                self.emit(points);
+                self.emit_eof();
+            })
+            .catch(function(err) {
+                self.trigger('error', new Error(err.message));
```

Comment:
Yeah! Let's do that after https://trello.com/c/OZ9hDJJa/115-standard-error-objects-for-all-repos.